hw/net/virtio-net.c

```diff
@@ -515,6 +515,15 @@ static RxFilterInfo *virtio_net_query_rxfilter(NetClientState *nc)
     return info;
 }
 
+static void virtio_net_allow_vhost(VirtIONet *n, bool allow)
+{
+    int i;
+    for (i = 0; i < n->max_queues; i++) {
+        NetClientState *nc = qemu_get_subqueue(n->nic, i)->peer;
+        nc->vhost_net_disabled = !allow;
+    }
```

Comment:
I wonder whether we can simply add a check in virtio_net_vhost_status() like:

if (!get_vhost_net(nc->peer))
        return;
if (!n->vhost_disabled)
        return;

Reply:
Of course we can. But I would prefer to make things more clear, i.e. make get_vhost_net() always return meaningful result, even in set_feature() after the vhost_disabled is set. Otherwise people can rely on get_vhost_net() and call its methods which potentially can do something that we do not expect. For example, in migration flow callbacks etc.